src/dotnet/Chat.UI.Blazor/Components/ChatSearch.razor

```diff
@@ -0,0 +1,28 @@
+<SearchBox IsCompact="true" IsProgressEnabled="true" MaxLength="@Constants.Chat.MaxSearchCriteriaLength"
+           Triggered="OnSearchTriggered" Cancelled="OnSearchCancelled"/>
+
+@code{
+    private ChatEntry? _lastFoundEntry;
+
+    [Inject] private IChats Chats { get; init; } = null!;
+    [Inject] private Session Session { get; init; } = null!;
+    [Inject] private UIEventHub UIEventHub { get; init; } = null!;
+    [Inject] private ChatUI ChatUI { get; init; } = null!;
+
+    [CascadingParameter] public Chat Chat { get; set; } = null!;
+
+    private async Task OnSearchTriggered(string text) {
+        var chatId = Chat.Id;
+        var nextEntry = await Chats.FindNext(Session, chatId, _lastFoundEntry?.Id, text, CancellationToken.None);
+        if (nextEntry == null)
+            return;
+
+        await UIEventHub.Publish(new NavigateToChatEntry(nextEntry.Id));
+        ChatUI.HighlightedChatEntryId.Value = nextEntry.Id;
+        _lastFoundEntry = nextEntry;
+    }
+
+    private void OnSearchCancelled() {
```

Comment:
Can we start using expression bodied members when appropriate?

Reply:
done